README.md

```diff
@@ -98,69 +136,103 @@ into the element. This is the only method that must be implemented by subclasses
+  nodes that must be directly acted upon, for example in `update()`.
+
+  * `updateComplete`:  Returns a Promise that resolves when the element has completed
+  updating that resolves to a boolean value that is `true` if the element completed the
+  update without triggering another update. This can happen if a property is set in
+  `update()` after the call to `super.update()` for example. This getter can be
+  implemented to await additional state. For example, it is sometimes useful to
+  await a rendered element before fulfilling this promise. To do this, first
+  await `super.updateComplete` then any subsequent state.
+
+  * `invalidate`: Call to request the element to asynchronously update regardless
+  of whether or not any property changes are pending. This should only be called
+  when an element should update based on some state not stored in properties,
+  since setting properties automically calls `invalidate`.
+
+  * `invalidateProperty(name, oldValue)` (protected): Triggers an invalidation for
+  a specific property. This is useful when manually implementing a property setter.
+  Call `invalidateProperty` instead of `invalidate` to ensure that any configured
+  property options are honored.
+
+  * `createRenderRoot()` (protected): Implement to customize where the
   element's template is rendered by returning an element into which to
   render. By default this creates a shadowRoot for the element.
   To render into the element's childNodes, return `this`.
 
-  * `_firstRendered()`: Called after the element DOM is rendered for the first time.
-
-  * `_shouldRender(props, changedProps, prevProps)`: Implement to control if rendering
-  should occur when property values change or `invalidate` is called.
-  By default, this method always returns true, but this can be customized as
-  an optimization to avoid rendering work when changes occur which should not be rendered.
-
-  * `_render(props)`: Implement to describe the element's DOM using `lit-html`. Ideally,
-  the `_render` implementation is a pure function using only `props` to describe
-  the element template. This is the only method that must be implemented by subclasses.
-
-  * `_didRender(props, changedProps, prevProps)`: Called after element DOM has been rendered.
-  Implement to directly control rendered DOM. Typically this is not needed as `lit-html`
-  can be used in the `_render` method to set properties, attributes, and
-  event listeners. However, it is sometimes useful for calling methods on
-  rendered elements, for example focusing an input:
-  `this.shadowRoot.querySelector('input').focus()`.
-
-  * `renderComplete`: Returns a promise which resolves after the element next renders.
-
-  * `_requestRender`: Call to request the element to asynchronously re-render regardless
-  of whether or not any property changes are pending.
+## Advanced: Update Lifecycle
+
+* When the element is first connected or a property is set (e.g. `element.foo = 5`)
```

Comment:
I find the nesting here to make this visually more complicated than it is. Also the "thens", "Notes" and other connector words.

Reply:
Rewrote